File: bwell-swift-ios/bwell-swift-ios/HealthSync/HealthMetricIcon.swift
//
//  HealthMetricIcon.swift
//  bwell-swift-ios
//
//  A small, hand-picked SF Symbol lookup for metric/body-system card icons -
//  ported in spirit (not literally) from ui-platform's mfe-devices, which
//  picks an icon per LOINC code from a remote-config JSON with a generic
//  fallback (see lib/metrics/adapters.ts's matchIconToVital). This demo has
//  no such remote config, so this matches on keywords in the metric's
//  display name instead - approximate, not a LOINC registry port. Anything
//  unmatched falls back to a generic trending-chart icon, same shape as the
//  real app's own fallback behavior.
//

import Foundation

enum HealthMetricIcon {
    private static let genericFallback = "chart.line.uptrend.xyaxis"

    /// (keyword, SF Symbol) pairs, checked in order against the lowercased
    /// title - first match wins. Order matters: more specific keywords
    /// (e.g. "blood pressure") are listed before generic ones they'd
    /// otherwise also match (e.g. "blood").
    private static let keywordIcons: [(keyword: String, symbol: String)] = [
        ("blood pressure", "heart.text.square.fill"),
        ("heart rate", "heart.fill"),
        ("heart", "heart.fill"),
        ("step", "figure.walk"),
        ("distance", "figure.walk.motion"),
        ("exercise", "figure.run"),
        ("activity", "flame.fill"),
        ("calorie", "flame.fill"),
        ("energy", "flame.fill"),
        ("sleep", "moon.zzz.fill"),
        ("weight", "scalemass.fill"),
        ("body mass", "scalemass.fill"),
        ("oxygen", "lungs.fill"),
        ("respiratory", "lungs.fill"),
        ("temperature", "thermometer"),
        ("glucose", "drop.fill"),
        ("blood", "drop.fill"),
    ]

    static func symbolName(for title: String?) -> String {
        guard let title, !title.isEmpty else { return genericFallback }
        let lowercased = title.lowercased()
        return keywordIcons.first { lowercased.contains($0.keyword) }?.symbol ?? genericFallback
    }
}

File: bwell-kotlin-android/app/src/main/java/com/bwell/sampleapp/activities/ui/healthsync/HealthMetricIcon.kt
package com.bwell.sampleapp.activities.ui.healthsync

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Air
import androidx.compose.material.icons.filled.Bedtime
import androidx.compose.material.icons.filled.Bloodtype
import androidx.compose.material.icons.filled.DirectionsRun
import androidx.compose.material.icons.filled.DirectionsWalk
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.LocalFireDepartment
import androidx.compose.material.icons.filled.MonitorHeart
import androidx.compose.material.icons.filled.MonitorWeight
import androidx.compose.material.icons.filled.ShowChart
import androidx.compose.material.icons.filled.Thermostat
import androidx.compose.ui.graphics.vector.ImageVector

/**
 * A small, hand-picked icon lookup for metric/body-system card icons -
 * ported from Swift's HealthMetricIcon.swift. Matches on keywords in the
 * metric's display name (this demo has no LOINC-code-to-icon remote config
 * like ui-platform's real one) - approximate, not a LOINC registry port.
 * Anything unmatched falls back to a generic trending-chart icon, same
 * shape as the real app's own fallback behavior.
 */
object HealthMetricIcon {
    private val genericFallback = Icons.Filled.ShowChart

    /** Checked in order - more specific keywords before generic ones they'd otherwise also match. */
    private val keywordIcons: List<Pair<String, ImageVector>> = listOf(
        "blood pressure" to Icons.Filled.MonitorHeart,
        "heart rate" to Icons.Filled.Favorite,
        "heart" to Icons.Filled.Favorite,
        "step" to Icons.Filled.DirectionsWalk,
        "distance" to Icons.Filled.DirectionsWalk,
        "exercise" to Icons.Filled.DirectionsRun,
        "activity" to Icons.Filled.LocalFireDepartment,
        "calorie" to Icons.Filled.LocalFireDepartment,
        "energy" to Icons.Filled.LocalFireDepartment,
        "sleep" to Icons.Filled.Bedtime,
        "weight" to Icons.Filled.MonitorWeight,
        "body mass" to Icons.Filled.MonitorWeight,
        "oxygen" to Icons.Filled.Air,
        "respiratory" to Icons.Filled.Air,
        "temperature" to Icons.Filled.Thermostat,
        "glucose" to Icons.Filled.Bloodtype,
        "blood" to Icons.Filled.Bloodtype,
    )

    fun iconFor(title: String?): ImageVector {
        if (title.isNullOrEmpty()) return genericFallback
        val lowered = title.lowercase()
        return keywordIcons.firstOrNull { (keyword, _) -> lowered.contains(keyword) }?.second ?: genericFallback
    }
}
